devU-client/src/components/pages/courseAssignmentsListPage.tsx

```diff
@@ -1,62 +1,66 @@
-import React,{useEffect,useState} from 'react'
+import React, { useEffect, useState } from 'react'
 
 import PageWrapper from 'components/shared/layouts/pageWrapper'
-import {Assignment} from 'devu-shared-modules'
+import { Assignment } from 'devu-shared-modules'
 import RequestService from 'services/request.service'
 import ErrorPage from './errorPage'
 import LoadingOverlay from 'components/shared/loaders/loadingOverlay'
 import { Link, useParams } from 'react-router-dom'
 
 import styles from './courseAssignmentsListPage.scss'
+import { useAppSelector } from '../../redux/hooks'
 
 
 const CourseAssignmentsListPage = () => {
 
-    const { courseId } = useParams<{courseId: string}>()
-    const [error, setError] = useState(null)
-    const [loading, setLoading] = useState(true)
-    const [assignments, setAssignments] = useState(new Array<Assignment>())
-
-    useEffect(() => {
-        fetchData()
-      }, [])
-
-    const fetchData = async () => {
-        try {
-            const assignments = await RequestService.get(`/api/assignments/course/${courseId}`)
-            setAssignments(assignments)
-        }catch(error){
-            setError(error)
-        }finally{
-            setLoading(false)
-        }
+  const role = useAppSelector((store) => store.roleMode)
+  const { courseId } = useParams<{ courseId: string }>()
+  const [error, setError] = useState(null)
+  const [loading, setLoading] = useState(true)
+  const [assignments, setAssignments] = useState(new Array<Assignment>())
+
+  useEffect(() => {
+    fetchData()
+  }, [])
+
+  const fetchData = async () => {
+    try {
+      const assignments = await RequestService.get(`/api/assignments/course/${courseId}`)
+      setAssignments(assignments)
+    } catch (error) {
+      setError(error)
+    } finally {
+      setLoading(false)
     }
+  }
+
+  if (loading) return <LoadingOverlay delay={250} />
+  if (error) return <ErrorPage error={error} />
+
+  return (
+    <PageWrapper>
+      <div className={styles.header}>
+        <h1>Course Assignments List Page</h1>
+
+        <div>
+          {( role.isInstructor() &&
```

Comment:
The course assignments page has effectively been deprecated on the latest version of dev, with the course detail page now listing the course's assignments. This toggle won't see use on dev and would be useful on the new details page.